### DISCUSSION TITLE
Is the detection algorithm identical for MQTT messages and for Alerts/Detections ?

### DISCUSSION THREAD

**Discussion by philippedegeer** [blakeblackshear/frigate#13500] [Ask A Question]
### What is your question?

I have two instances of Frigate at two different house and they both exhibit the following behaviour:

Within HA, I regularly receive "Person Occupancy" ON and then OFF MQTT messages separated by one second, while there is no alert or detection in Frigate. Also, my Frigate+ panel does not show any detection. This is an issue for me as I trigger a siren based on these MQTT messages to HA.

In the documentation it says that detection is based on the median value of the last three scores. Can someone confirm that this is true for both MQTT messages and alerts/detection for recording ? 

Thanks.

**Comment by NickM-27** [comment]
yes it's true, but there are many other checks that are used for alerts / detections. If you want to match that behavior you should use the /reviews MQTT topic, otherwise you will need to apply filters to the /events topic

**Reply by philippedegeer** [reply]
Thanks Nick, this clarifies the question and what I should do next. Philippe

**Comment by jonofmac** [comment]
Bump: Thanks for asking this question. I have an automation that also triggers on occupancy and it gets triggered frequently but I almost never get an event/detection.

I would like to understand the differences. The occupancy sensor is very convenient to trigger automations off of, so I'd like to understand the difference, too

**Reply by NickM-27** [reply]
https://docs.frigate.video/integrations/home-assistant#if-i-am-detecting-multiple-objects-how-do-i-assign-the-correct-binary_sensor-to-the-camera-in-homekit

**Reply by jonofmac** [reply]
In my case, I'm actually getting true positives on the occupancy sensor but NOT getting an event/detection, and I'm trying to chase down why that is. So it appears that some of these "additional checks" for the events prevent frigate from saving it as a detection. 

I wasn't sure if it's "inertia" being set to 3 (I tried 1), my minimum score is at 0.7 on the frigate plus models.

**Reply by NickM-27** [reply]
Best bet is to save a recording copy and rerun it to see what happens on the debug view
